### DISCUSSION TITLE
OffScreenWindow is miss in gtk4, which one I could use to replace in gtk4

### DISCUSSION THREAD

**Discussion by Sherlock-Holo** [gtk-rs/gtk4-rs#1655] [Q&A]
https://gtk-rs.org/gtk3-rs/stable/latest/docs/gtk/struct.OffscreenWindow.html

I draw a widget to `OffscreenWindow`, then call `.draw()` to draw the widget to a foreign cairo context

but I found `OffScreenWindow` is miss in gtk4, so how to draw a widget to a foreign cairo context in gtk4?

**Comment by bilelmoussaoui** [comment]
I suppose you could use a GtkWidgetPaintable, create a custom gtk::Snapshot, snapshot the paintable there (the following code can be used for inspiration https://gitlab.gnome.org/World/design/app-icon-preview/-/blob/master/src/widgets/project_previewer.rs?ref_type=heads#L117) and I guess you could take the gdk::Texture and draw it content on a cairo context. There might better/other ways, but I would recommend asking gtk developers about it.
